src/WingetCreateCLI/Models/DscModels/SettingsResourceObject.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateCLI.Models.DscModels;
+
+using Microsoft.WingetCreateCLI.Properties;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+/// <summary>
+/// Represents a settings resource object used in DSC operations.
+/// </summary>
+public class SettingsResourceObject : BaseResourceObject
+{
+    /// <summary>
+    /// Defines the action type for full settings application.
+    /// </summary>
+    public const string ActionFull = "Full";
+
+    /// <summary>
+    /// Defines the action type for partial settings application.
+    /// </summary>
+    public const string ActionPartial = "Partial";
+
+    /// <summary>
+    /// Gets or sets the settings for the resource object.
+    /// </summary>
+    [JsonProperty("settings", Required = Required.Always)]
+    public JObject Settings { get; set; }
+
+    /// <summary>
+    /// Gets or sets the action to be performed on the settings resource object.
+    /// </summary>
+    [JsonProperty("action", NullValueHandling = NullValueHandling.Ignore)]
+    public string Action { get; set; }

```

Comment:
"= ActionPartial" to be explicit?

Reply:
nit: if we don't want default value, type should be "string?"

I know we have not turned on stylecop for this project, but we will one day..

Reply:
We don't have nullables enabled in the project, so the default for a string is `null`.
Since we don't have nullable reference enabled, using `string?` will cause a CS8632 warning.